src/chunk.c

```diff
@@ -4937,3 +5159,218 @@ ts_chunk_get_osm_slice_id(int32 chunk_id, int32 time_dim_id)
 	const int slice_id = ds->fd.id;
 	return slice_id;
 }
+
+/*
+ * Initialization and access method for ChunkVec. This needs to be extended to support additional
+ * operations.
+ */
+static ChunkVec *
+chunk_vec_expand(ChunkVec *chunks, uint32 new_capacity)
+{
+	if (chunks != NULL && chunks->capacity > new_capacity)
+		return chunks;
+
+	if (chunks == NULL)
+		chunks = palloc(CHUNK_VEC_SIZE(new_capacity));
+	else
+		chunks = repalloc(chunks, CHUNK_VEC_SIZE(new_capacity));
+
+	chunks->capacity = new_capacity;
+
+	return chunks;
+}
+
+ChunkVec *
+ts_chunk_vec_create(int32 num_chunks)
+{
+	ChunkVec *chunks = chunk_vec_expand(NULL, num_chunks);
+
+	chunks->num_chunks = 0;
+	return chunks;
+}
+
+ChunkVec *
+ts_chunk_vec_sort(ChunkVec **chunks)
```

Comment:
Function never called? Is it used or case just not hit? In case of the latter, should try to hit that case.

Reply:
Good catch. To be similar to the sorted output for regular `time` based queries, we will call this even when we use creation time. Fixed.